scripts/run_presubmit_checks.sh

```diff
@@ -47,11 +61,47 @@ python $(dirname $0)/pre_commit_linter.py || exit 1
 echo 'Linting passed.'
 echo ''
 
-# Run frontend unit tests.
-echo 'Running frontend unit tests'
-source $(dirname $0)/run_frontend_tests.sh --run-minified-tests=true || exit 1
-echo 'Frontend tests passed.'
-echo ''
+# Read arguments from the command line.
+for i in "$@"
+do
+case $i in
+    -b=*|--branch=*)
+    ORIGIN_BRANCH=${i#*=}
+    shift
+    ;;
+esac
+done
+
+CURRENT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
+# If the current branch exists on remote origin, MATCHED_BRANCH_NUM=1
+# else MATCHED_BRANCH_NUM=0
+MATCHED_BRANCH_NUM=$(git ls-remote --heads origin $CURRENT_BRANCH | wc -l)
+# Set the origin branch to develop if it's not specified.
+if [ -n "$ORIGIN_BRANCH" ]; then
+  BRANCH=$ORIGIN_BRANCH
+elif [ $MATCHED_BRANCH_NUM == 1 ]; then
+  BRANCH=origin/$CURRENT_BRANCH
+else
+  BRANCH=develop
+fi
+
+FRONT_END_DIR='core/templates/dev/head'
+
+echo "Comparing the current branch with $BRANCH"
+
+if git diff --cached --name-only --diff-filter=ACM ${BRANCH} | grep ${FRONT_END_DIR} --quiet
```

Comment:
I don't have super-good bash-fu, but should there be any parentheses/brackets here?

Reply:
I'm also not so good at bash. According to my search. using a bracket might need further change and it works well without bracket. See [Stack Exchange](https://unix.stackexchange.com/questions/48535/can-grep-return-true-false-or-are-there-alternative-methods)